engine/engine.go

```diff
@@ -168,13 +158,93 @@ func Start(ctx context.Context, startOpts *Config, fn StartCallback) error {
 			}
 
 			return bkgw.NewResult(), nil
-		}, startOpts.RawBuildkitStatus)
+		}, solveCh)
 		return err
 	})
 
 	return eg.Wait()
 }
 
+func handleSolveEvents(startOpts *Config, ch chan *bkclient.SolveStatus) error {
+	eg := &errgroup.Group{}
+	readers := []chan *bkclient.SolveStatus{}
+
+	// Dispatch events to raw listener
+	if startOpts.RawBuildkitStatus != nil {
+		readers = append(readers, startOpts.RawBuildkitStatus)
+	}
+
+	// Print events to the console
+	if startOpts.LogOutput != nil {
+		ch := make(chan *bkclient.SolveStatus)
+		readers = append(readers, ch)
+		eg.Go(func() error {
+			warn, err := progressui.DisplaySolveStatus(context.TODO(), "", nil, startOpts.LogOutput, ch)
+			for _, w := range warn {
+				fmt.Fprintf(os.Stderr, "=> %s\n", w.Short)
```

Comment:
nit: pre-existing issue but I think the warning should be sent to startOpts.LogOutput instead of stderr